### PR TITLE
MARP-148 S4HANA connector - update rest client, example and document

### PR DESCRIPTION
**Pull request by phhung-axonivy**

### CODE REVIEW

**Comment by phhung-axonivy** [comment]
Hi @ivy-sgi 
Could you help me review the file S4HANA-product/README.md in this pull request?

File: S4HANA/config/rest-clients.yaml [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
     Features:
     - ch.ivyteam.ivy.rest.client.mapper.JsonFeature
     - ch.ivyteam.ivy.rest.client.authentication.HttpBasicAuthenticationFeature
     Properties:
       JSON.Deserialization.FAIL_ON_UNKNOWN_PROPERTIES: 'false'
+      username: ${ivy.var.s4HanaConnector.username}
+      password: ${ivy.var.s4HanaConnector.password}
     OpenAPI:
-      SpecUrl: file:///C:/Users/hb/git/S4HANA/S4HANA/lib/generated/rest/openapi.json
+      SpecUrl: file:///C:/Users/phhung/Downloads/API_BUSINESS_PARTNER%20(1).json
```
Comment (ndkhanh-axonivy) [inline S4HANA/config/rest-clients.yaml:14]: Please delete this one

File: S4HANA-product/README.md [hunk 2]
```diff
@@ -3,16 +3,29 @@
 This connector shows an easy way to connect to the SAP S4HANA database.
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:3]: What about:
This SAP S4/HANA connector easily connects to the SAP S4/HANA database. This connector:
- focuses on the business partner domain
- establishes a connection to its database to show all business partners together with their email addresses
- needs a tryout account for the [SAP Demo API](https://api.sap.com/api/API_BUSINESS_PARTNER/tryout)
- supports you with an easy-to-copy demo implementation to reduce your integration effort

File: S4HANA-product/README.md [hunk 3]
```diff
@@ -3,16 +3,29 @@
 This connector shows an easy way to connect to the SAP S4HANA database. 
 
 The focus here is on the business partner domain. In the example, a connection to the database is established and all business partners are listed together with their email addresses.
-Please note: In the example, a connection to [SAP Demo API](https://api.sap.com/api/BusinessPartner_APIs/tryout) is established, i.e. it is necessary that you first create an account here.
+Please note: In the example, a connection to [SAP Demo API](https://api.sap.com/api/API_BUSINESS_PARTNER/tryout) is established, i.e. it is necessary that you first create an account here.
 
 
 ## Demo
 
 Demo tries to connect to SAP and gets list of business partners
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:11]: This demo example to connects to the database of all business partners in SAP S4/HANA.

File: S4HANA-product/README.md [hunk 4]
```diff
@@ -3,16 +3,29 @@
...
+1. Start Process to get a list of business partners
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:12]: 1. Start the process "Get list of business partners"

File: S4HANA-product/README.md [hunk 5]
```diff
@@ -3,16 +3,29 @@
...
+![get-a-list-of-business-partners](images/get-a-list-of-business-partners.png)
+2. List of business partners are displayed in the table
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:14]: 2. A list of business partners are displayed in the table

File: S4HANA-product/README.md [hunk 6]
```diff
@@ -3,16 +3,29 @@
...
+![list-of-business-partners](images/list-of-business-partners.png)
 
 ## Setup
 
-Please set the global variables to the SAP needs. Maybe you need to add a certificate to connect to SAP via SSL.
+1. Setup Environment for S/4HANA Business Partner API to get an account that contains host(baseUrl), username and password to use.
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:19]: 1. Set up a test environment for SAP S4/HANA Business Partner API to get an account containing the host(baseUrl), username, and password.

File: S4HANA-product/README.md [hunk 7]
```diff
@@ -3,16 +3,29 @@
...
+
+2. Please set the global variables to the SAP needs. Maybe you need to add a certificate to connect to SAP via SSL.
```
Comment (andreasbalsiger) [inline S4HANA-product/README.md:21]: 2. Set the global variables to SAP's needs. You may need to add a certificate to connect to SAP via SSL.

**Review by andreasbalsiger — COMMENTED** [review]
please have a look at my edits.

**Comment by andreasbalsiger** [comment]
@ivy-sgi have a look at my edits.

**Review by ivy-sgi — APPROVED** [review]
I like your changes - approved :)

**Comment by ivy-sgi** [comment]
Hi @phhung-axonivy From our side (@andreasbalsiger and me) this connector could be published now. If everything is ok from your side just hit the button :)

**Pull request merged**